application/controllers/artifacts_controller.py

```diff
@@ -11,27 +11,29 @@
 from webargs.flaskparser import use_args
 
 from .application_controller import ApplicationController
-from ..extensions import socketio
-from ..socketio_parser import use_args as socketio_args
-from ..responders import no_content, respond_with
 from ..error_handling.es_connection import check_es_connection
 from ..errors import NotFound
-from ..models.artifact import Artifact
+from ..extensions import socketio
+from ..models.artifact_builder import ArtifactBuilder
 from ..recognition.image_recognition import ImageRecognizer
+from ..responders import no_content, respond_with
+from ..socketio_parser import use_args as socketio_args
 from ..synonyms.synonyms import SynonymGenerator
 from ..validators import artifacts_validator
 
+
 @socketio.on("SYNCHRONIZED_SEARCH")
 @socketio_args(artifacts_validator.search_args())
 def synchronized_search(params):
     """ Called from client when presentation mode is on """
-    artifacts = Artifact.search(params)
+    artifacts = ArtifactBuilder.search(params)
```

Comment:
Why is it called ArtifactBuilder now?

Reply:
The endstate I want that class to be is an ArtifactBuilder. It should encapsulate how the relations between artifacts, tags and teams are connected. Currently it also does the job of syncing with elastic search which is something I'll be working on next. So the class only has one responsibility left.